test/unit/cluster_resource_discovery_test.rb

```diff
@@ -17,6 +17,15 @@ def test_global_resource_kinds_success
     end
   end
 
+  def test_prunable_resources
```

Comment:
Maybe it's not necessary but one could also refute_includes Node and Namespace yeah? Seems like an unlikely regression but might be nice to be explicit about.